implants/lib/eldritch/src/sys/get_reg_impl.rs

```diff
@@ -0,0 +1,83 @@
+use anyhow::Result;
+use starlark::{values::{dict::Dict, Heap, Value}, collections::SmallMap};
+
+
+pub fn get_reg(starlark_heap: &Heap, reghive: String, regpath: String) -> Result<Dict>  {
+    
+    let res: SmallMap<Value, Value> = SmallMap::new();
+    let mut tmp_res = Dict::new(res);
+    
+
+    #[cfg(not(target_os = "windows"))]
+        return Err(anyhow::anyhow!("This OS isn't supported by the get_reg function. Only windows systems are supported"));
+
+    #[cfg(target_os = "windows")]{
+        use winreg::{{enums::*}, RegKey};
+        //Accepted values for reghive :
+        //HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE, HKEY_USERS, HKEY_PERFORMANCE_DATA, HKEY_PERFORMANCE_TEXT, HKEY_PERFORMANCE_NLSTEXT, HKEY_CURRENT_CONFIG, HKEY_DYN_DATA, HKEY_CURRENT_USER_LOCAL_SETTINGS
+
+        let ihive: isize = match reghive.as_ref() {
+            "HKEY_CLASSES_ROOT" => HKEY_CLASSES_ROOT,
+            "HKEY_CURRENT_USER" => HKEY_CURRENT_USER,
+            "HKEY_LOCAL_MACHINE" => HKEY_LOCAL_MACHINE,
+            "HKEY_USERS" => HKEY_USERS,
+            "HKEY_PERFORMANCE_DATA" => HKEY_PERFORMANCE_DATA,
+            "HKEY_PERFORMANCE_TEXT" => HKEY_PERFORMANCE_TEXT,
+            "HKEY_PERFORMANCE_NLSTEXT" => HKEY_PERFORMANCE_NLSTEXT,
+            "HKEY_CURRENT_CONFIG" => HKEY_CURRENT_CONFIG,
+            "HKEY_DYN_DATA" => HKEY_DYN_DATA,
+            "HKEY_CURRENT_USER_LOCAL_SETTINGS" => HKEY_CURRENT_USER_LOCAL_SETTINGS,
+            _ => return Err(anyhow::anyhow!("RegHive can only be one of the following values - HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, HKEY_LOCAL_MACHINE, HKEY_USERS, HKEY_PERFORMANCE_DATA, HKEY_PERFORMANCE_TEXT, HKEY_PERFORMANCE_NLSTEXT, HKEY_CURRENT_CONFIG, HKEY_DYN_DATA, HKEY_CURRENT_USER_LOCAL_SETTINGS ")),
+            
+        };
+        
+        
+        let hive = RegKey::predef(ihive);
+        let subkey = hive.open_subkey(regpath)?;
+
+        
+        for (key, val) in subkey.enum_values().map(|x| x.unwrap()) {
```

Comment:
Don't use unwrap in eldritch functions this will cause the agent to crash unrecoverably. Instead use a match to handle the error or ? To bubble the error up. 
If it's an option you're unwrapping I believe the anyhow context function will allow you to add an error string and use ? to bubble the error or get the result.